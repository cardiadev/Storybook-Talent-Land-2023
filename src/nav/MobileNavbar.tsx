import classNames from "classnames";
import React, { FC } from "react";
import { Typography } from "../Typography";
import { renderLogo } from "./SideNav";

export interface MobileNavbarProps {
  open: boolean;
  toggleOpen: () => void;
}

export const MobileNavbar: FC<MobileNavbarProps> = ({ open, toggleOpen }) => (
  <div className="z-40 flex items-center justify-between h-20 py-6 bg-white shadow-md dark:bg-gray-900 px-9">
    {renderLogo()}
    <Typography variant="h6" customWeight="medium" className="select-none">
      My Travel App
    </Typography>
    <svg
      data-testid="toggleOpen"
      width="32"
      height="32"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
      className={classNames(
        "text-gray-500 transition-all duration-100 ease-out cursor-pointer stroke-current hover:text-gray-900 dark:text-white dark:hover:text-gray-200",
        {
          "opacity-0": open,
        },
      )}
      onClick={toggleOpen}
    >
      <path
        d="M4 16H22.6667M4 8H28M4 24H28"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
);
