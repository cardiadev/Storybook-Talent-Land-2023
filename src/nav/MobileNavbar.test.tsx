import * as React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MobileNavbar } from "./MobileNavbar";

afterEach(cleanup);

describe("MobileNavbar", () => {
  it("Renders MobileNavbar open correctly", () => {
    const { asFragment } = render(<MobileNavbar open toggleOpen={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders MobileNavbar closed correctly", () => {
    const { asFragment } = render(
      <MobileNavbar open={false} toggleOpen={() => {}} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Can open/close MobileNavbar", () => {
    let open = true;

    const { rerender, getByTestId } = render(
      <MobileNavbar
        open={open}
        toggleOpen={() => {
          open = !open;
        }}
      />,
    );

    // Close
    fireEvent.click(getByTestId("toggleOpen"));

    rerender(
      <MobileNavbar
        open={open}
        toggleOpen={() => {
          open = !open;
        }}
      />,
    );

    expect(open).toEqual(false);

    // Open
    fireEvent.click(getByTestId("toggleOpen"));

    rerender(
      <MobileNavbar
        open={open}
        toggleOpen={() => {
          open = !open;
        }}
      />,
    );

    expect(open).toEqual(true);
  });
});
