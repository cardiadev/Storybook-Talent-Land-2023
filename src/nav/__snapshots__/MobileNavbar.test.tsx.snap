// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavbar Renders MobileNavbar closed correctly 1`] = `
<DocumentFragment>
  <div
    class="z-40 flex items-center justify-between h-20 py-6 bg-white shadow-md dark:bg-gray-900 px-9"
  >
    <img
      alt="logo"
      class="w-8 h-8 select-none"
      src="https://res.cloudinary.com/tailwindcss/image/upload/v1634916081/Logo_Icon_dq276z.png"
    />
    <h6
      class="text-h6 font-medium select-none text-black dark:text-white"
    >
      My Travel App
    </h6>
    <svg
      class="text-gray-500 transition-all duration-100 ease-out cursor-pointer stroke-current hover:text-gray-900 dark:text-white dark:hover:text-gray-200"
      data-testid="toggleOpen"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 16H22.6667M4 8H28M4 24H28"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`MobileNavbar Renders MobileNavbar open correctly 1`] = `
<DocumentFragment>
  <div
    class="z-40 flex items-center justify-between h-20 py-6 bg-white shadow-md dark:bg-gray-900 px-9"
  >
    <img
      alt="logo"
      class="w-8 h-8 select-none"
      src="https://res.cloudinary.com/tailwindcss/image/upload/v1634916081/Logo_Icon_dq276z.png"
    />
    <h6
      class="text-h6 font-medium select-none text-black dark:text-white"
    >
      My Travel App
    </h6>
    <svg
      class="text-gray-500 transition-all duration-100 ease-out cursor-pointer stroke-current hover:text-gray-900 dark:text-white dark:hover:text-gray-200 opacity-0"
      data-testid="toggleOpen"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 16H22.6667M4 8H28M4 24H28"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </div>
</DocumentFragment>
`;
